pkgs/applications/science/electronics/vivado/default.nix

```diff
@@ -0,0 +1,127 @@
+{ stdenv
+, lib
+, coreutils
+, fetchurl
+, patchelf
+, procps
+, makeWrapper
+, requireFile
+, ncurses5
+, zlib
+, libuuid
+, libSM
+, libICE
+, libX11
+, libXrender
+, libxcb
+, libXext
+, libXtst
+, libXi
+, glib
+, freetype
+, gtk2
+}:
+
+stdenv.mkDerivation rec {
+  pname = "vivado";
+  version = "2017.2";
+
+  buildInputs = [
+    patchelf
+    procps
+    ncurses5
+    makeWrapper
+    coreutils
+  ];
+
+  inherit ncurses5;
+
+  # requireFile prevents rehashing each time, which saves time during
+  # rebuilds.
+  src = requireFile rec {
+    name = "Xilinx_Vivado_SDK_2017.2_0616_1.tar.gz";
+    message = ''
+      This nix expression requires that ${name} is already part of the store.
+      Login to Xilinx, download from
+      https://www.xilinx.com/support/download.html,
+      rename the file to ${name}, and add it to the nix store with
+      "nix-prefetch-url file:///path/to/${name}".
+    '';
+    sha256 = "06pb4wjz76wlwhhzky9vkyi4aq6775k63c2kw3j9prqdipxqzf9j";
+  };
+
+  patchPhase = ''
+    patchShebangs .
+    patchelf --set-interpreter "$(cat $NIX_CC/nix-support/dynamic-linker)" \
+         tps/lnx64/jre/bin/java
+  '';
+
+  dontBuild = true;
+
+  installPhase = ''
+    # generate an installation config file
```

Comment:
From where did you get this?

Reply:
This, along with pretty everything else, I took from the original derivation (see [this](https://github.com/lukaslaobeyer/nix-fpgapkgs/issues/1)). Do you want something changed from it? Some of these settings could probably exposed so that someone can change them more easily.